x-pack/plugin/data-streams/src/main/java/org/elasticsearch/xpack/datastreams/rest/RestDataStreamsStatsAction.java

```diff
@@ -29,6 +29,12 @@ public String getName() {
         );
     }
 
+    @Override
+    public boolean allowSystemIndexAccessByDefault() {
+        // Does not actually access system indices, but does resolve `_all`, which can cause confusing deprecation warnings.
```

Comment:
this feels like a hack to me; I wonder if we can find a way to not need this for items that resolve _all to do certain things :|

Reply:
It's definitely a hack, and I'd love to not do this. I'll dig in a bit and see if there's a better way to handle this case.